src/net_processing.h

```diff
@@ -9,6 +9,8 @@
 #include "net.h"
 #include "validationinterface.h"
 
+/** Maximum number of outstanding CMPCTBLOCK requests for the same block. */
+static const int MAX_CMPCTBLOCKS_INFLIGHT_PER_BLOCK = 2;
```

Comment:
What do you think the upper limit is before it would generally cause a negative impact? Maybe just have an upper limit like we do for max outbound connections.